packages/react-app/pages/admin/newBatches.jsx

```diff
@@ -0,0 +1,373 @@
+      return rows;
+    }
+
+    if (filterValue === "allBatches") {
+      return rows.filter(row => row.values[id]?.batch?.number);
+    }
+
+    return rows.filter(row => {
+      const rowValue = row.values[id];
+      return rowValue !== undefined ? isInBatch(rowValue, filterValue) : true;
+    });
+  };
+
+  useEffect(() => {
+    async function fetchBuilders() {
+      setIsLoadingBuilders(true);
+      const fetchedBuilders = await axios.get(serverUrl + serverPath);
+
+      const processedBuilders = fetchedBuilders.data
+        .filter(builder => builder.ens !== "austingriffith.eth")
+        .map(builder => ({
+          builder,
+          status: builder.status,
+          stream: builder,
+          builds: builder.builds?.length || 0,
+          socials: builder,
+          userCreated: builderCreated(builder),
+        }));
+
+      setBuilders(processedBuilders);
+      setIsLoadingBuilders(false);
+    }
+
+    fetchBuilders();
+  }, [serverUrl]);
+
+  const BatchFilterComponent = useMemo(() => {
+    const Component = ({ column }) => (
+      <BatchColumnFilter filterValue={column.filterValue} setFilter={column.setFilter} builders={builders} />
+    );
+    Component.displayName = "BatchFilterComponent";
+    return Component;
+  }, [builders]);
+
+  const columns = useMemo(
+    () => [
+      getColumn(
+        "Builder",
+        "builder",
+        ({ value }) => <BuilderAddressCell builder={value} mainnetProvider={mainnetProvider} />,
+        { disableSortBy: true, Filter: EnsColumnFilter, filter: ensFiltering },
+      ),
+      getColumn("Status", "status", ({ value }) => <BuilderStatusCell status={value} />, { disableSortBy: true }),
+      getColumn("Builds", "builds", ({ value }) => <BuilderBuildsCell buildCount={value} />, { sortDescFirst: true }),
+      getColumn(
+        "Stream",
+        "stream",
+        ({ value }) => (value?.graduated?.status ? "" : <StreamTableCell builder={value} />),
+        { disableFilters: true, Filter: BatchFilterComponent, filter: batchFiltering },
+      ),
+      getColumn("Socials", "socials", ({ value }) => <BuilderSocialLinksCell builder={value} isAdmin={isAdmin} />, {
+        disableSortBy: true,
+      }),
+      getColumn("User Created", "userCreated", ({ value }) => <DateWithTooltip timestamp={value} />, {
+        sortAscFirst: true,
+      }),
+    ],
+    [BatchFilterComponent, mainnetProvider, isAdmin],
+  );
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    prepareRow,
+    page,
+    canPreviousPage,
+    canNextPage,
+    pageOptions,
+    pageCount,
+    gotoPage,
+    nextPage,
+    previousPage,
+    setPageSize,
+    state: { pageIndex, pageSize },
+  } = useTable(
+    {
+      columns,
+      data: builders,
+      initialState: { pageIndex: 0, pageSize: 25, sortBy: useMemo(() => [{ id: "stream", desc: true }], []) },
+    },
+    useFilters,
+    useSortBy,
+    usePagination,
+  );
```

Comment:
Maybe for another PR... but I think it would be useful to have some kind of editing (even multiple at the same time)? So you can mark people as graduated, change batch number, etc.

You could take /admin/fund as an example

But yeah, maybe for a future PR if needed! (don't want to optimize early)

Reply:
I agree that would be nice, let me work on that afterwards.